### PR TITLE
Updates new structure

### PR DESCRIPTION
**Pull request by bago2k4**

### CODE REVIEW

**Comment by bago2k4** [comment]
[New Structure](https://trello.com/c/AcouEW18/632-new-structure)

**Comment by belucid** [comment]
@bago2k4 

Early feedback on this before you asked for it:

- [ ] Tests failing
- [ ] Mobile sizing of topics in update issue: https://opencompanyhq.slack.com/files/stuart/F31TVA0CD/slack_for_ios_upload._null_
- [ ] Still see the no updates message before it's replaced w/ the updates: https://files.slack.com/files-pri/T06SBMH60-F31TBUKC4/slack_for_ios_upload._null_
- [ ] Should see Updates nav, priors and list on public companies that you aren't logged in to

**Comment by belucid** [comment]
Reviewing.

File: scss/partials/_navbar.scss [hunk 1]
```diff
@@ -80,33 +106,43 @@ nav.oc-navbar {
...
-      }
-
       div.user-avatar {
         float: right;
       }
     }
   }
+
+  div.oc-navbar-bottom {
+    backgorund-color: blue;
```
Comment (belucid) [inline scss/partials/_navbar.scss]: Typo here.

File: src/open_company_web/components/create_update.cljs [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            [open-company-web.components.topics-columns :refer (topics-columns)]
+            [open-company-web.components.su-preview-dialog :refer (su-preview-dialog)]
+            [clojure.data :as cd]))
+
+(defn ordered-topics-list []
+  (let [topics (sel [:div.create-update-topics-list :div.oc-active])
+        topics-list (for [topic topics] (.data (js/jQuery topic) "topic"))]
+    (vec (remove nil? topics-list))))
+
+(defn remove-pinned [data]
```
Comment (belucid) [inline src/open_company_web/components/create_update.cljs]: I don't love the construction of this at all.

That's a lot of code for no comment(s) and/or docstring.

Use of `loop`, `recur` is a bit of a code smell to me. I've found there is almost always a better, more functional construction.

Lastly, this seems prone to an infinite loop if any of this data/logic goes wrong.

Strongly suggest we refactor this.

File: src/open_company_web/components/create_update.cljs [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                all-keys
+                (inc idx))))))
+
+(defn patch-stakeholder-update [owner]
+  (let [title  (om/get-state owner :su-title)
+       topics (om/get-state owner :su-topics)]
+    (api/patch-stakeholder-update {:title (or title "")
+                                   :sections topics})))
+
+(defn setup-sortable [owner]
```
Comment (belucid) [inline src/open_company_web/components/create_update.cljs]: Comment/docstring here would be good.

File: src/open_company_web/actions.cljs [hunk 4]
```diff
@@ -136,26 +136,33 @@
...
-  (let [protocol (.. js/document -location -protocol)
-        host     (.. js/document -location -host)
-        su-url   (str protocol "//" host (oc-urls/stakeholder-update slug (utils/su-date-from-created-at su-date) su-slug))]
+  (let [su-url   (oc-urls/stakeholder-update slug (utils/su-date-from-created-at su-date) su-slug)]
     (-> db
       (assoc-in (dispatcher/latest-stakeholder-update-key slug) su-url)
       (dissoc :loading))))
 
-(defmethod dispatcher/action :stakeholder-update [db [_ {:keys [slug update-slug response]}]]
+(defmethod dispatcher/action :stakeholder-update [db [_ {:keys [slug update-slug response load-company-data]}]]
```
Comment (belucid) [inline src/open_company_web/actions.cljs]: Comment/docstring here would be good.

File: src/open_company_web/actions.cljs [hunk 5]
```diff
@@ -419,4 +426,17 @@
...
+  [db [_]]
+  (assoc db :mobile-menu-open (not (:mobile-menu-open db))))
+
+(defmethod dispatcher/action :reset-su-list
+  [db [_]]
+  (let [slug (router/current-company-slug)
+        su-list-key (dispatcher/su-list-key slug)
+        latest-su-key (dispatcher/latest-stakeholder-update-key slug)]
+    (-> db
+      (update-in (butlast su-list-key) dissoc (last su-list-key))
```
Comment (belucid) [inline src/open_company_web/actions.cljs]: these 2 are expressions are fairly complicated, a comment here about what and why this is happening would be good.

File: src/open_company_web/components/company_dashboard.cljs [hunk 6]
```diff
@@ -42,6 +41,7 @@
      :columns-num (responsive/columns-num)})
 
   (did-mount [_]
+    (utils/after 100 #(set! (.-scrollTop (.-body js/document)) 0))
```
Comment (belucid) [inline src/open_company_web/components/company_dashboard.cljs]: comment here on why this is happening (from a user/UX perspective) would be good

File: src/open_company_web/components/prior_updates.cljs [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+(def desktop-width 500)
+
+(defn- half-offset [pixels]
+  (str "-" (Math/round (* 0.5 pixels)) "px"))
+
+(defn update-click [link e]
+  (utils/event-stop e)
+  (router/nav! link))
+
+(defn load-prior-updaes-if-needed []
```
Comment (belucid) [inline src/open_company_web/components/prior_updates.cljs]: typo in function name.

can be private?

File: src/open_company_web/components/prior_updates.cljs [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+
+(def link-formatter (f/formatter "yyyy-MM-dd"))
+(def human-formatter (f/formatter "MMMM d, yyyy"))
+
+(def desktop-width 500)
+
+(defn- half-offset [pixels]
+  (str "-" (Math/round (* 0.5 pixels)) "px"))
+
+(defn update-click [link e]
```
Comment (belucid) [inline src/open_company_web/components/prior_updates.cljs]: can be private?

**Comment by belucid** [comment]
@bago2k4 comments are in above.

**Pull request merged**